storage_v2/factoryadapter/reader.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package factoryadapter
+
+import (
+	"context"
+	"errors"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	spanstore_v1 "github.com/jaegertracing/jaeger/storage/spanstore"
+	"github.com/jaegertracing/jaeger/storage_v2/spanstore"
+)
+
+var errV1ReaderNotAvailable = errors.New("spanstore.Reader is not a wrapper around v1 reader")
+
+type TraceReader struct {
+	spanReader spanstore_v1.Reader
+}
+
+func GetV1Reader(reader spanstore.Reader) (spanstore_v1.Reader, error) {
+	if tr, ok := reader.(*TraceReader); ok {
+		return tr.spanReader, nil
+	}
+	return nil, errV1ReaderNotAvailable
+}
+
+func NewTraceReader(spanReader spanstore_v1.Reader) *TraceReader {
+	return &TraceReader{
+		spanReader: spanReader,
+	}
+}
+
+func (*TraceReader) GetTrace(_ context.Context, _ pcommon.TraceID) (ptrace.Traces, error) {
+	panic("not implemented")
```

Comment:
at some point we would need to implement these methods, because as soon as we start upgrading query subcomponents to use v2 storage api directly (e.g. the apiv3 handler, the easiest to upgrade since it already returns OTLP formats) they will be blocked on this implementation for v1-only backends.

Reply:
@yurishkuro what would these implementations look like?

Reply:
you would call the underlying v1 reader to retrieve the trace and then convert the result into OTLP data model